applications/solvers/solids4Foam/Make/files

```diff
@@ -1,3 +1,5 @@
+ddtCorr/EulerDdtScheme.C
+ddtCorr/backwardDdtScheme.C
```

Comment:
@ztukovic , I presume these ddtCorr files are added here so they will take the place of the standard OpenFOAM classes; is that correct?

I wonder if this behaviour is system or compiler specific. I wonder if renaming the files (e.g. EulerDdtS4FScheme.C) would work; but if these classes are to be used by OpenFOAM system classes then this may be the only solution.

Reply:
On my macOS using the clang compiler, it gives an error when compiling the solids4Foam solver:
```
Undefined symbols for architecture arm64:
  "Foam::fvPatchField<Foam::Vector<double>>::patchConstructorTablePtr_", referenced from:
      Foam::fvPatchField<Foam::Vector<double>>::New(Foam::word const&, Foam::fvPatch const&, Foam::DimensionedField<Foam::Vector<double>, Foam::volMesh> const&) in EulerDdtScheme.o
      Foam::fvPatchField<Foam::Vector<double>>::New(Foam::word const&, Foam::fvPatch const&, Foam::DimensionedField<Foam::Vector<double>, Foam::volMesh> const&) in backwardDdtScheme.o
  "Foam::fvPatchField<Foam::Vector<double>>::disallowGenericFvPatchField", referenced from:
      Foam::fvPatchField<Foam::Vector<double>>::New(Foam::fvPatch const&, Foam::DimensionedField<Foam::Vector<double>, Foam::volMesh> const&, Foam::dictionary const&) in EulerDdtScheme.o
      Foam::fvPatchField<Foam::Vector<double>>::New(Foam::fvPatch const&, Foam::DimensionedField<Foam::Vector<double>, Foam::volMesh> const&, Foam::dictionary const&) in backwardDdtScheme.o
...
```

If I remove `tCorr/EulerDdtScheme.C` and `ddtCorr/backwardDdtScheme.C` from `Make/files` then it compiles fine. It seems that the compiler expects the entire EulerDdtScheme.C and backwardDdtScheme.C classes to be defined in these files (rather than one function). I will add those and see if it helps.

Reply:
Hmnn... I cannot get this to compile with Clang.

@ztukovic, do you have any suggestions? Some options could be:
- create EulerDdt**Vector**Scheme and backwardDdt**Vector**Scheme classes, which provide vector template specialisations with the required changes (I guess this should work, but I am unsure of how to call the makeFvDdtTypeScheme macros)
- only allow this if the compiler is GCC, otherwise, give a warning during compilation and an error in the relevant cases
- maybe there is some way to move this change to optionalFixes so that the user can decide if they want to make this change in their main installation

Reply:
I think solution with optionalFixes is optimal

Reply:
I have moved ddtScheme files into optionalFixes folder and modified Allcheck script